packages/babel-helpers/src/helpers.js

```diff
@@ -861,8 +861,13 @@ helpers.arrayWithHoles = helper("7.0.0-beta.0")`
 helpers.iterableToArray = helper("7.0.0-beta.0")`
   export default function _iterableToArray(iter) {
     if (
-      Symbol.iterator in Object(iter) ||
-      Object.prototype.toString.call(iter) === "[object Arguments]"
+      Array.isArray(iter)
+      || typeof iter === 'string'
+      || (typeof Symbol === 'function' && Symbol.iterator in Object(iter))
+      || (iter && 'length' in iter)
+      || Object.prototype.toString.call(iter) === "[object Map]"
```

Comment:
I'm not sure I think including these is worth it. If it's doesn't pass `Array.isArray(iter)`, then we can ask people to load a symbol polyfill or convert it to an array before using it as an iterable, since it really isn't one unless they've done one of those two things anyway.

Reply:
es6-shim creates a Map and Set that work with Array.from, without Symbol.iterator - without this change, Airbnb remains broken on environments without symbols.

Reply:
My issue here is that i'd consider the use of `Array.from` an implementation detail. If `Symbol` doesn't exist, then your item isn't iterable. The fact that we support it for arrays is a necessity, but I don't see it that way for `Map` and `Set`. I'd expect you to call `Array.from` to convert to an array first.

Reply:
Anyway, now it will not work with `es6-shim` because it does not support `@@toStringTag`.